### PR TITLE
Add support for properly functioning navbar submenus

### LINKED ISSUE #721: multiple level submenu in navbar?
It seems it is not possible to add multiple level submenu in navbar to render a website. If so, it will be really nice to add this features for complicated website.

This is my example _navbar.yml. A arrow is shown in the second level of menu, but cannot open it.

``` yaml
title: WIKI
type: inverse
left:
    - text: "Home"
      href: index.html
    - text: Menu1
      menu:
        - text: "Menu 1.1"
          href: menu-1-1.html
        - text: "Menu 1.2"
          href: menu-1-2.html
        - text: "Menu 1.3"
          menu: 
            - text: "Menu 1.3.1"
              href: menu-1-3-1.html
            - text: "Menu 1.3.2"
              href: menu-1-3-2.html 


```

### PR DESCRIPTION
**Pull request by rich-iannone**
This fix is for replacing the class of a submenu item (3rd level down) such that it gets the `dropdown-submenu` class.

Fixes #721

### CODE REVIEW

**Comment by rich-iannone** [comment]
For the `_site.yml` of:

```
name: "site"
navbar:
  title: "My Website"
  left:
    - text: "Home"
      href: index.html
    - text: "About"
      href: about.html
      menu:
        - text: "Sub_1"
          href: sub1.html
        - text: "Sub_2"
          menu:
            - text: "Sub_2_below"
              href: sub2.html
```

this change targets the second to last item (with text `Sub_2`) so that the `sub2.html` is available and clickable. Here is a screenshot:

submenu-fix

This uses a CSS path that targets any item with this type of relationship.

This script generates files used to generate the site (and does the site render also):

```r

index <- '---
title: "My Website"
output: html_document
---

Hello, Website!'

about <- '---
title: "About This Website"
output: html_document
---

More about this website.'

sub_part_1 <- '---
title: "About This Website"
output: html_document
---

sub_1: This is right below the About page.'

sub_part_2 <- '---
title: "About This Website"
output: html_document
---

sub_2: This is 2 levels below the About page.'

write(index, file = "index.Rmd")
write(about, file = "about.Rmd")
write(sub_part_1, file = "sub1.Rmd")
write(sub_part_2, file = "sub2.Rmd")

rmarkdown::render("index.Rmd")
rmarkdown::render("about.Rmd")
rmarkdown::render("sub1.Rmd")
rmarkdown::render("sub2.Rmd")

render_site()
```

**Comment by rich-iannone** [comment]
The submenu now functions properly and jQuery was not used in this final solution. Here is a screenshot of the menu/submenu:

submenu

Some CSS rules had to be added to 'restore' support for the `dropdown-submenu` CSS class (removed in Bootstrap 3.0 RC, see https://stackoverflow.com/questions/18023493/bootstrap-dropdown-sub-menu-missing for additional context).

The R code changes occurred entirely in the `navbar_links_tags()` function. By checking the depth, we can distinguish between a top-level menu element or a menu element below top level. Then, we can apply the `dropdown` or `dropdown-submenu` class to the `li` element.

**Comment by rich-iannone** [comment]
Forgot to mention it earlier... thanks @jcheng5 for the assistance on this!

File: inst/rmd/h/default.html [hunk 1]
```diff
@@ -215,6 +215,49 @@
...
+.dropdown-submenu {
+    position:relative;
+}
+.dropdown-submenu>.dropdown-menu {
+    top:0;
+    left:100%;
+    margin-top:-6px;
+    margin-left:-1px;
+    -webkit-border-radius:0 6px 6px 6px;
+    -moz-border-radius:0 6px 6px 6px;
```
[the remark this answers is not in the document]
Comment (rich-iannone) [reply inst/rmd/h/default.html]: Nice, I'll take those lines out.

File: inst/rmd/h/default.html [hunk 2]
```diff
@@ -215,6 +215,49 @@
   padding-top: $header_padding$px;
   margin-top: -$header_padding$px;
 }
+
+.dropdown-submenu {
+    position:relative;
```
[the remark this answers is not in the document]
Comment (rich-iannone) [reply inst/rmd/h/default.html]: Sorry for missing this. Will do!

**Comment by yihui** [comment]
We also need a NEWS entry for this feature (or perhaps bug fix).

**Pull request merged**